service/application/src/main/java/com/solace/maas/ep/event/management/agent/command/CommandManager.java

```diff
@@ -103,12 +103,18 @@ public void configPush(CommandMessage request) {
                     log.error("Error executing command", e);
                     setCommandError(command, e);
                 }
+                if (bundle.getExitOnFailure() && command.getResult().getStatus() == JobStatus.error
+                        && !Boolean.TRUE.equals(command.getIgnoreResult())) {
+                    break;
+                }
```

Comment:
sonarlint really wants all Boolean boxed values to be unboxed safer accounting for a null value like the `Boolean.TRUE.equals()` here.
I'd like to at least change this around so that it reads `bundle.getExitOnFailure() &&  JobStatus.error.equals(command.getResult().getStatus())`. Sonarlint is happy with `(Boolean.TRUE.equals(bundle.getExitOnFailure() &&  JobStatus.error.equals(command.getResult().getStatus()))` but it seems too verbose to me.

Also, I thought getResult() could be null if `ignoreResult` is true, yeah? This would be typical for import actions. The error result would be passed on to the apply. Looks like the logic here is always setting the result on the command that failed. I don't think this was mentioned in the ADR so I want to add what we decide to it.

Reply:
Right, good point about the boolean logic/null checking. I'll re-work this logic.
I'll look at the terraform plugin logic to make sure we're not setting the result if the `ignoreResult` flag is set.